src/main/java/uk/gov/ch/oauth/tokens/SessionSignInModifier.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.ch.oauth.tokens;
+
+import java.util.Map;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.SessionKeys;
+import uk.gov.companieshouse.session.model.AccessToken;
+import uk.gov.companieshouse.session.model.UserProfile;
+
+public class SessionSignInModifier {
+
+    @SuppressWarnings("unchecked")
+    // This is necessary as the original data on the session is untyped, but expected to be of the correct types
+    static Map<String, Object> updateSignIn(final Object sInf, final Object sio) {
+        final Map<String, Object> original = (Map<String, Object>) sInf;
+        final Map<String, Object> extras = (Map<String, Object>) sio;
+        original.putAll(extras);
+        return original;
+    }
+
+    public void alterSessionData(Session session, AccessToken accessToken,
+            UserProfile userProfileResponse) {
```

Comment:
Why are we passing in session, when we're not using it in this method. I think we can do without.

Reply:
I chose to pass in session here because it simplifies code on the calling side and it clearly separates the overloaded methods into ones that are interacting with the objects and ones that interact with the maps that represent data.